ReactNativeClient/lib/services/KeymapService.ts

```diff
@@ -0,0 +1,284 @@
+		{ accelerator: 'Shift+Cmd+L', command: 'focusElementNoteList' },
+		{ accelerator: 'Shift+Cmd+N', command: 'focusElementNoteTitle' },
+		{ accelerator: 'Shift+Cmd+B', command: 'focusElementNoteBody' },
+		{ accelerator: 'Option+Cmd+S', command: 'toggleSidebar' },
+		{ accelerator: 'Cmd+L', command: 'toggleVisiblePanes' },
+		{ accelerator: 'Cmd+0', command: 'zoomActualSize' },
+		{ accelerator: 'Cmd+E', command: 'startExternalEditing' },
+		{ accelerator: 'Option+Cmd+T', command: 'setTags' },
+		{ accelerator: 'Cmd+G', command: 'gotoAnything' },
+		{ accelerator: 'F1', command: 'help' },
+	];
+
+	static readonly defaultKeymap_OtherPlatforms: KeymapItem[] = [
+		{ accelerator: 'Ctrl+N', command: 'newNote' },
+		{ accelerator: 'Ctrl+T', command: 'newTodo' },
+		{ accelerator: 'Ctrl+S', command: 'synchronize' },
+		{ accelerator: 'Ctrl+P', command: 'print' },
+		{ accelerator: 'Ctrl+Q', command: 'quit' },
+		{ accelerator: 'Ctrl+Alt+I', command: 'insertTemplate' },
+		{ accelerator: 'Ctrl+C', command: 'textCopy' },
+		{ accelerator: 'Ctrl+X', command: 'textCut' },
+		{ accelerator: 'Ctrl+V', command: 'textPaste' },
+		{ accelerator: 'Ctrl+A', command: 'textSelectAll' },
+		{ accelerator: 'Ctrl+B', command: 'textBold' },
+		{ accelerator: 'Ctrl+I', command: 'textItalic' },
+		{ accelerator: 'Ctrl+K', command: 'textLink' },
+		{ accelerator: 'Ctrl+`', command: 'textCode' },
+		{ accelerator: 'Ctrl+Shift+T', command: 'insertDateTime' },
+		{ accelerator: 'F6', command: 'focusSearch' },
+		{ accelerator: 'Ctrl+F', command: 'showLocalSearch' },
+		{ accelerator: 'Ctrl+Shift+S', command: 'focusElementSideBar' },
+		{ accelerator: 'Ctrl+Shift+L', command: 'focusElementNoteList' },
+		{ accelerator: 'Ctrl+Shift+N', command: 'focusElementNoteTitle' },
+		{ accelerator: 'Ctrl+Shift+B', command: 'focusElementNoteBody' },
+		{ accelerator: 'F10', command: 'toggleSidebar' },
+		{ accelerator: 'Ctrl+L', command: 'toggleVisiblePanes' },
+		{ accelerator: 'Ctrl+0', command: 'zoomActualSize' },
+		{ accelerator: 'Ctrl+E', command: 'startExternalEditing' },
+		{ accelerator: 'Ctrl+Alt+T', command: 'setTags' },
+		{ accelerator: 'Ctrl+,', command: 'config' },
+		{ accelerator: 'Ctrl+G', command: 'gotoAnything' },
+		{ accelerator: 'F1', command: 'help' },
+	];
+
+	static readonly defaultKeymap = shim.isMac()
+		? KeymapService.defaultKeymap_macOS
+		: KeymapService.defaultKeymap_OtherPlatforms;
+
+	constructor() {
+		super();
+		this.keymap = KeymapService.initKeymap();
+
+		const customKeymapPath = `${Setting.value('profileDir')}/keymap-desktop.json`;
+		if (fs.existsSync(customKeymapPath)) {
+			try {
+				const customKeymap = JSON.parse(fs.readFileSync(customKeymapPath, 'utf-8'));
+				this.setKeymap(customKeymap);
+			} catch (err) {
+				const msg = err.message ? err.message : '';
+				throw new Error(
+					`KeymapService: Failed to override keymap at ${customKeymapPath}!\n\n${msg}`
+				);
+			}
+		}
```

Comment:
This should be done in a loadKeymap which will take a keymap file path as input. That way you won't need a dependency to Setting and it will make the class dependencies more obvious.

Reply:
Okay, I'll move this into a function.